lib/xnvme_be_linux_nvme.c

```diff
@@ -283,6 +284,41 @@ xnvme_be_linux_nvme_cmd_admin(struct xnvme_cmd_ctx *ctx, void *dbuf, size_t dbuf
 	return 0;
 }
 
+int
+xnvme_be_linux_nvme_cmd_pseudo(struct xnvme_cmd_ctx *ctx, void *XNVME_UNUSED(dbuf),
+			       size_t XNVME_UNUSED(dbuf_nbytes), void *XNVME_UNUSED(mbuf),
+			       size_t XNVME_UNUSED(mbuf_nbytes))
+{
+	int err;
+	int ioctl_req = 0;
+	struct xnvme_be_linux_state *state = (void *)ctx->dev->be.state;
+
+	switch (ctx->cmd.common.opcode) {
+	case XNVME_SPEC_PSEUDO_OPC_SUBSYSTEM_RESET:
+		ioctl_req = NVME_IOCTL_SUBSYS_RESET;
+		break;
+
+	case XNVME_SPEC_PSEUDO_OPC_CONTROLLER_RESET:
+		ioctl_req = NVME_IOCTL_RESET;
+		break;
+
+	case XNVME_SPEC_PSEUDO_OPC_NAMESPACE_RESCAN:
+		ioctl_req = NVME_IOCTL_RESCAN;
+		break;
+	}
+
+	err = ioctl(state->fd, ioctl_req);
+	if (err < 0) {
+		XNVME_DEBUG("FAILED: ioctl() err: %d", err);
+		return err;
+	}
+
+	ctx->cpl.status.sc = 0;
+	ctx->cpl.status.sct = 0;
```

Comment:
Do we need to set these explicitly? Perhaps we should use `ioctl_wrap()` instead of `ioctl()`?